packages/fluentui/e2e/e2eApi.ts

```diff
@@ -35,47 +35,78 @@ export class E2EApi {
     return await this.page.hover(selector);
   };
 
-  public getAttributeValue = async (selector: string, attr) => {
-    return this.page.$eval(selector, (el, attribute) => el.getAttribute(attribute), attr);
+  public hasComputedStyle = async (
+    selector: string,
+    property: keyof CSSStyleDeclaration,
+    value: string,
+  ): Promise<void> => {
+    await this.page.waitForFunction(
+      (selector, property, value) => {
+        return window.getComputedStyle(document.querySelector(selector))[property] === value;
+      },
+      { timeout: 10 * 1000 },
+      selector,
+      property,
+      value,
+    );
   };
 
   public getPropertyValue = async (selector: string, prop) => {
     return this.page.$eval(selector, (el, prop) => el[prop], prop);
   };
 
-  public count = async (selector: string) => {
-    return (await this.page.$$(selector)).length;
+  public count = async (selector: string, count: number): Promise<void> => {
+    await this.page.waitForFunction(
+      (selector, count) => {
+        return document.querySelectorAll(selector).length === count;
+      },
+      { timeout: 10 * 1000 },
+      selector,
+      count,
+    );
   };
 
-  public exists = async (selector: string) => {
-    return (await this.count(selector)) > 0;
+  public exists = async (selector: string): Promise<void> => {
+    await this.page.waitForSelector(selector, { timeout: 10 * 1000 });
   };
 
-  public clickOn = async (selector: string) => await (await this.getElement(selector)).click();
-
-  public clickOnPosition = async (selector: string, x: number, y: number) => {
-    const elementHandle = await this.getElement(selector);
-    const boundingBox = await elementHandle.boundingBox();
-
-    await this.page.mouse.click(Math.round(boundingBox.x) + x, Math.round(boundingBox.y) + y);
+  public hidden = async (selector: string): Promise<void> => {
+    await this.page.waitForSelector(selector, { hidden: true, timeout: 10 * 1000 });
   };
 
-  public textOf = async (selector: string) => {
-    const element = await this.getElement(selector);
-    return await (await element.getProperty('textContent')).jsonValue();
+  public clickOn = async (selector: string): Promise<void> => {
+    await (await this.getElement(selector)).click();
   };
 
-  public focusOn = async (selector: string) => await (await this.getElement(selector)).focus();
+  public textOf = async (selector: string, text: string): Promise<void> => {
```

Comment:
Should these helper assert functions return some kind of meaningful error message instead of only throwing a generic puppeteer exception ?

Reply:
It will be great, I will think how to implement it in more obvious way